vangogh-sample/src/main/java/com/example/vangogh/menu/MainMenuAdapter.java

```diff
@@ -0,0 +1,63 @@
+package com.example.vangogh.menu;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.example.vangogh.R;
+
+import java.util.List;
+
+public class MainMenuAdapter extends ArrayAdapter<MainMenuItem> {
+
+    private OnMainMenuItemClickListener listener;
```

Comment:
Should be marked `@Nullable`.

Reply:
Or maybe inject a `@NonNull` listener through the adapter's constructor? It's not necessary in this setup that the listener is optional.